wire-compiler/src/main/java/com/squareup/wire/model/Options.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.model;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.squareup.protoparser.OptionElement;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+/**
+ * A set of options declared on a message declaration, field declaration, enum declaration, enum
+ * constant declaration, service declaration, RPC method declaration, or proto file declaration.
+ * Options values may be arbitrary protocol buffer messages, but must be valid protocol buffer
+ * messages.
+ */
+public final class Options {
+  private final ProtoTypeName optionType;
+  private final String packageName;
+  private final ImmutableList<OptionElement> optionElements;
+  private ImmutableMap<WireField, Object> map;
+
+  public Options(ProtoTypeName optionType, String packageName, List<OptionElement> elements) {
+    this.optionType = optionType;
+    this.packageName = packageName;
+    this.optionElements = ImmutableList.copyOf(elements);
+  }
+
+  public String packageName() {
+    return packageName;
+  }
+
+  /**
+   * Returns a map with the values for these options. Map values may be either a single entry, like
+   * {@code {default: "5"}}, or more sophisticated, with nested maps and lists.
```

Comment:
nit: Is Javadoc smart enough here? I think you'll probably need to use `<code>` HTML tags due to the curly braces.